rtv/content.py

```diff
@@ -123,7 +125,14 @@ def strip_praw_submission(sub):
         data['subreddit'] = strip_subreddit_url(sub.permalink)
         data['flair'] = flair
         data['url_full'] = sub.url
-        data['url'] = ('selfpost' if is_selfpost(sub.url) else sub.url)
+
+        if reddit_link.match(sub.url):
+            stripped_url = reddit_link_no_host.search(sub.url).group()
+            stripped_comments = reddit_link_no_host.search(sub.permalink).group()
+            data['url'] = ('selfpost' if stripped_url == stripped_comments
+                    else 'x-post')
+        else:
+            data['url'] = sub.url
```

Comment:
I think that this the logic here is a kind of confusing to follow. I was thinking of ways to potentially clean it up and came up with this -

The values that we have are:
``data['permalink']`` - Link to the submission comments
``data['url_full']`` - Link that the submission points to
``data['url']`` - What is displayed on the screen

So in keeping with this convention, we should really add a new field for the post type. That way you don't have to hijack the ``data['url']`` for x-posts. I think Brobin was eluding to this also.

```python
if data['permalink'].split('/r/')[-1] == data['url_full'].split('/r/')[-1]:
    data['url_type'] = 'selfpost'
    data['url'] = 'selfpost'

elif reddit_link.match(data['url_full']):
    data['url_type'] = 'x-post'
    data['url'] = 'x-post via {}'.format(data['subreddit'])

else:
    data['url_type'] = 'external'
    data['url'] = data['url_full']
```

This also simplifies subreddit.py because you can go back to printing ``data['url']`` directly.